pkg/admission/validator/shoot.go

```diff
@@ -99,6 +100,17 @@ func (s *shoot) Validate(ctx context.Context, new, old client.Object) error {
 		return fmt.Errorf("invalid cloud credentials: %v", err)
 	}
 
+	// The Kubernetes version is the version of the CSI migration, where we stopped using the in-tree providers.
+	// see: pkg/webhook/controlplane/ensurer.go
+	k8sVersionLessThan19, err := version.CompareVersions(shoot.Spec.Kubernetes.Version, "<", "1.19")
+	if err != nil {
+		return err
+	}
+
+	if k8sVersionLessThan19 && s.isUsingApplicationCredentials(secret) {
```

Comment:
Should we not rather check this in `ExtractCredentials()`?
And pass `k8sVersionLessThan19` then to `ExtractCredentials()`.
Then we could also adapt the unit test there.

Reply:
We agreed not to move it to `ExtractCredentials()` as it is used in different places.